stack.cabal

```diff
@@ -219,7 +219,7 @@ library
                    , vector-binary-instances
                    , yaml >= 0.8.10.1
                    , zlib >= 0.5.4.2 && < 0.7
-                   , deepseq >= 1.4
+                   , deepseq >= 1.3 && < 1.5
```

Comment:
Seems like there was a reason for the lower bound: 51813ffe (I found this with git blame)

Regarding the upper bound, I don't think we really need it, because `stack` uses `stack sdist --pvp-bounds both` (and hopefully soon the extension discussed in #2262) for publishing, so the releases already have upper bounds.